lib/appConfig.ts

```diff
@@ -1,32 +1,45 @@
-import { StrategyOptions } from 'passport-github'
+const { URL } = require("url");
 
 export interface AppConfig {
-  isDevelopment: boolean
-  hostingURL: string
-  github: StrategyOptions
-}
+  isDevelopment: boolean;
+  hostingURL: string;
 
-const getOAuthUrls: (
-  hostName: string,
-  app: string
-) => { callbackURL: string } = (hostName: string, app: string) => ({
-  // Alternatively, use `[app].ts` filenames for paramaterized urls
-  callbackURL: `${hostName}/api/auth/callback/${app}`,
-})
+  getCredentialsForOAuthProvider(
+    provider: string
+  ): {
+    provider: string;
+    clientID: string;
+    clientSecret: string;
+  };
+  getOAuthCallbackUrl(provider: string, host?: string): string;
+}
+const isDevelopment = process.env.NODE_ENV !== "production";
+const hostingURL = process.env.HOSTING_URL || "http://localhost:3000";
 
-const isDevelopment = process.env.NODE_ENV !== 'production'
-const hostingURL = process.env.HOSTING_URL || 'http://localhost:3000'
+const getOAuthCallbackUrl = (provider: string, host?: string) => {
```

Comment:
I can see we are relying on string in the input parameter of URL object, what do we think about using string literal types fro providers to scope it down?

```suggestion
type Providers = 'google' | 'github';

const getOAuthCallbackUrl = (provider: Providers, host?: string) => {
  ...
```